crates/bevy_ecs/src/system/commands.rs

```diff
@@ -174,6 +177,20 @@ impl CommandsInternal {
         self
     }
 
+    pub fn spawn_with(&mut self, component: impl Component) -> &mut Self {
+        let entity = self
+            .entity_reserver
+            .as_ref()
+            .expect("entity reserver has not been set")
+            .reserve_entity();
+        self.current_entity = Some(entity);
```

Comment:
Should we place this in a private method `Commands::new_entity` to ensure entity reservation and its error message stays consistent between `spawn_with` and `spawn_with_bundle`?

Reply:
That's sensible...if we use this sort of API.